### PR TITLE
Tweak illegal `Copy` impl message

### PR DESCRIPTION
**Pull request by compiler-errors**
The phrase "may not" can both mean "is not able to" and "possibly does not". Disambiguate this by just using "cannot".
@Lokathor expressed being annoyed by this [here](https://twitter.com/Lokathor/status/1633200313544089602?s=20).

Also drive-by fix for this extremely noisy message: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6a37275bc810f7846bfe191845b7d11d.

r? diagnostics

### CODE REVIEW

File: library/core/src/marker.rs [hunk 1]
```diff
@@ -324,7 +324,7 @@ pub trait StructuralEq {
 /// attempt to derive a `Copy` implementation, we'll get an error:
 ///
 /// ```text
-/// the trait `Copy` may not be implemented for this type; field `points` does not implement `Copy`
+/// the trait `Copy` cannot be implemented for this type; field `points` does not implement `Copy`
```
Comment (compiler-errors) [inline library/core/src/marker.rs:327]: Lol, this is just a documentation change.

File: compiler/rustc_hir_analysis/src/coherence/builtin.rs [hunk 2]
```diff
@@ -94,7 +95,14 @@ fn visit_implementation_of_copy(tcx: TyCtxt<'_>, impl_did: LocalDefId) {
             let mut errors: BTreeMap<_, Vec<_>> = Default::default();
             let mut bounds = vec![];
 
+            let mut seen_tys = FxHashSet::default();
+
             for (field, ty, reason) in fields {
+                // Only report an error once per type.
```
Comment (Lokathor) [inline compiler/rustc_hir_analysis/src/coherence/builtin.rs:101]: so if more than one field isn't copy, it won't report all fields? am i understanding that right?

Comment (compiler-errors) [reply compiler/rustc_hir_analysis/src/coherence/builtin.rs:101]: It does not highlight subsequent fields with identical types.

Comment (Lokathor) [reply compiler/rustc_hir_analysis/src/coherence/builtin.rs:101]: oh once per *field type*, got it.

Comment (compiler-errors) [reply compiler/rustc_hir_analysis/src/coherence/builtin.rs:101]: This is actually unrelated to the example code you gave me, and was something that @fee1-dead had pointed out somewhere else a few days ago. It just came to mind when I was making the change.

Comment (fee1-dead) [reply compiler/rustc_hir_analysis/src/coherence/builtin.rs:101]: I think it was from @WaffleLapkin? Anyways thanks for working on this!

Comment (compiler-errors) [reply compiler/rustc_hir_analysis/src/coherence/builtin.rs:101]: oops, just remember seeing a screenshot of a really bad error message somewhere :rofl:

Comment (WaffleLapkin) [reply compiler/rustc_hir_analysis/src/coherence/builtin.rs:101]: The screenshot in question:


And ye, thanks for fixing this, errs <3

**Review by WaffleLapkin — APPROVED** [review]

**Pull request merged**